index.js

```diff
@@ -108,30 +86,38 @@ function getRoomFromSocket(socket_obj){
 // validates socket requests
 function validateSocket(socket_obj){
     const room_id = getRoomFromSocket(socket_obj)
-
+    const room = getGameFromSet(room_id)
     // adds socket to room if client is missing room and if room exists
     // will drop connections on server restart
-    if (room_id !== null && !socket_obj.rooms.has(room_id) && list_of_active_games.includes(room_id)){
+    if (room_id !== null && !socket_obj.rooms.has(room_id) && room !== null){
         socket_obj.join(room_id)
+        room.addPlayer(socket_obj.id)
+        return {room_id, room}
     }
     // drops connection if not validated
     else {
         socket_obj.disconnect(true)
+        return null
     }
-
-    return room_id
 }
 
+// there exists documentation for Socket IO Middleware
+// possible to implement before io.on connection
 io.on('connection', socket => {
-    const room_id = validateSocket(socket)
-    // TODO:
-    // here we should instantiate a game obj
-    // to track game state and stack history
+    const {room_id, room} = validateSocket(socket)
+
+    socket.emit('room_id', room_id)
+    socket.emit('player_number', room.getPlayerNumber(socket.id))
+    // needed to enable turn for room creator
+    io.to(room.getCurrentPlayer()).emit('enable_turn', true)
 
     // update users in room for new image
     socket.on('gameplay_stroke', data => {
```

Comment:
in this case data is the image data right ? 
so maybe jsut call it
`json_image_data`